integration/runtime_config_test.go

```diff
@@ -0,0 +1,96 @@
+//go:build requires_docker
+// +build requires_docker
+
+package integration
+
+import (
+	"bytes"
+	"context"
+	"fmt"
+	"os"
+	"path/filepath"
+	"testing"
+
+	"github.com/stretchr/testify/require"
+	"github.com/thanos-io/objstore/providers/s3"
+
+	"github.com/cortexproject/cortex/integration/e2e"
+	e2edb "github.com/cortexproject/cortex/integration/e2e/db"
+	"github.com/cortexproject/cortex/integration/e2ecortex"
+)
+
+func TestLoadRuntimeConfigFromStorageBackend(t *testing.T) {
+	s, err := e2e.NewScenario(networkName)
+	require.NoError(t, err)
+	defer s.Close()
+
+	require.NoError(t, copyFileToSharedDir(s, "docs/configuration/runtime-config.yaml", runtimeConfigFile))
+	require.NoError(t, copyFileToSharedDir(s, "docs/configuration/single-process-config-blocks-local.yaml", cortexConfigFile))
+
+	filePath := filepath.Join(e2e.ContainerSharedDir, runtimeConfigFile)
+	tests := []struct {
+		name  string
+		flags map[string]string
+	}{
+		{
+			name: "no storage backend provided",
+			flags: map[string]string{
+				"-runtime-config.file": filePath,
+			},
+		},
+		{
+			name: "filesystem as storage backend",
+			flags: map[string]string{
+				"-runtime-config.file":    filePath,
+				"-runtime-config.backend": "filesystem",
+			},
+		},
+	}
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			cortex := e2ecortex.NewSingleBinaryWithConfigFile("cortex-1", cortexConfigFile, tt.flags, "", 9009, 9095)
+			require.NoError(t, s.StartAndWaitReady(cortex))
+			require.NoError(t, s.Stop(cortex))
```

Comment:
Can we have a way to make sure what we load has the same content as the provided config file?
There is a runtime config API https://cortexmetrics.io/docs/api/#runtime-configuration.

Reply:
Yep, I added more test assertions in the latest commit